### ISSUE TITLE
binary resource uploading and sharing support (images, files, etc.)

### ISSUE THREAD

**Issue by doouding** [toeverything/blocksuite#4402]

**Comment by satoren** [comment]
I am facing the same problem. Although not efficient, I considered converting the blob to base64 and treating it as a key, but that was impossible because the BlobEngine set appears to return a key, but it is not used at all.
https://github.com/toeverything/blocksuite/pull/6937

Specifically, [this page ](https://try-blocksuite.vercel.app/)does not support image transfer.

@Flrande Do you have any ideas?

**Comment by Flrande** [comment]
> I am facing the same problem. Although not efficient, I considered converting the blob to base64 and treating it as a key, but that was impossible because the BlobEngine set appears to return a key, but it is not used at all. #6937
> 
> Specifically, [this page ](https://try-blocksuite.vercel.app/)does not support image transfer.
> 
> @Flrande Do you have any ideas?

BlockSuite Playground does not support Blob sharing for collaboration, as it is only meant for people to try out and debug. If you need to support the sharing of Blob resources, you'll need to implement your own [`BlobSource`](https://github.com/toeverything/blocksuite/pull/6937).

**Comment by satoren** [comment]
Thanks for the reply. 
> you'll need to implement your own https://github.com/toeverything/blocksuite/pull/6937.

I understand that. Even if we implement that, it seems like the only way to implement it is to require an external server, is that correct?

**Comment by satoren** [comment]
The following changes.
https://github.com/satoren/blocksuite/commit/99f0b351e66865cdb2cff789bfa98fbdcad5fd71

**Comment by Flrande** [comment]
I suggest having a server for storing Blob data, as transmitting it solely through Base64 may impose a significant burden on collaborative operations.

**Comment by satoren** [comment]
When I checked to see if the load was that high, I confirmed that the playground requires an additional 4 times the network bandwidth and CPU processing because it sends the yjs payload in a json array representation. This is certainly fatal.

**Comment by Flrande** [comment]
Yes, the implementation of Playground is very simple. You can refer to the code of AFFiNE for actual production.

https://github.com/toeverything/AFFiNE/tree/canary/packages/common/infra/src/sync/doc
